RecoParticleFlow/PFProducer/interface/PFMultiLinksIndex.h

```diff
@@ -0,0 +1,73 @@
+#ifndef MultiLinks_h
+#define MultiLinks_h
+
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElement.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <vector>
+#include <set>
+
+namespace reco {
+
+  struct pair_hash {
+    template <class T1, class T2>
+    std::size_t operator()(const std::pair<T1, T2>& p) const {
+      auto h1 = std::hash<T1>{}(p.first);
+      auto h2 = std::hash<T2>{}(p.second);
+      return h1 ^ h2;
+    }
+  };
+
+  //stores the links between PFElements that are found by KDTrees.
+  class PFMultiLinksIndex {
+  public:
+    PFMultiLinksIndex(size_t nelements) {
+      _linkdata[std::make_pair(reco::PFBlockElement::ECAL, reco::PFBlockElement::TRACK)].resize(nelements,
```

Comment:
```suggestion
      _linkdata[{reco::PFBlockElement::ECAL, reco::PFBlockElement::TRACK}].resize(nelements,
```
wouldn't this work as well (shorter)?

Reply:
ah nice, it did not occur to me!